libraries/lib-numeric-formats/NumericConverter.cpp

```diff
@@ -779,7 +779,7 @@ void NumericConverter::ParseFormatString(
       mFields[i].pos = pos;
 
       for(j=0; j<mFields[i].digits; j++) {
-         mDigits.push_back(DigitInfo(i, j, pos));
+         mDigits.push_back(DigitInfo { int(i), j, pos });
```

Comment:
I see this cast, and think, better that `i`, `j`, `pos` and the members of the structure be `size_t` -- unless -1 for invalid values is used somewhere?

Reply:
I kinda kept that cast intact (it was implicit previously). 

It is weird that old code (and the one moved to a new class) haven't really used the `DigitInfo::field`. Instead it tried to match the `pos` of the digits to get the corresponding field.

Reply:
I would even make the structure members `const size_t`

Then also, `NumericField::digits` should be `size_t`